Shopping- HTML layout

VaishaliRamesh-ng opened the pull request:
HTML layout for product, cart and home page for mobile tablet and desktop

VaishaliRamesh-ng committed:
Add files via upload

VaishaliRamesh-ng committed:
Second_commit with html layout for Cart, product

hiteshparashar4 commented on server.js:
```diff
@@ -0,0 +1,51 @@
+    var categories;
+    var products;
+    http.get('http://localhost:5000/banners',(resp)=>{
+        resp.on('data',(res)=>{
```
grab url from some external config or constant file. Don't hardcode

hiteshparashar4 commented on static/styles/banners.css:
```diff
@@ -0,0 +1,65 @@
+}
+@media (min-width: 1025px) {
+  body{
+    max-width: 80%;
```
dont try to directly update style of the body tag. Add a container div under body and update its styles if needed

hiteshparashar4 commented on static/styles/banners.css:
```diff
@@ -0,0 +1,65 @@
+} */
+
+.categories>ul>li>section>button{
+    background-color: #e91e63;
```
would appreciate if you could take care of the indentation. Add a space after/before every >. Add space before {

hiteshparashar4 commented on static/styles/cart.css:
```diff
@@ -0,0 +1,77 @@
+}
+body>div:nth-child(1){
+    background-color: black;
+    color: white;
```
same as above, dont directly update styles of body

hiteshparashar4 commented on static/styles/cart.css:
```diff
@@ -0,0 +1,77 @@
+    margin: 10px;;
+}
+button{
+    background-color: #e91e63;
```
styles for common elements like button span p etc must exist in separate file, say common.css

hiteshparashar4 commented on static/styles/cart.css:
```diff
@@ -0,0 +1,77 @@
+}
+footer>button{
+    width: 100%;
+}
```
use scss

hiteshparashar4 commented on static/styles/products.css:
```diff
@@ -0,0 +1,142 @@
+    object-fit: cover;
+}
+button{
+    width:100%;
```
button should look same across all webpages. So no need to add multiple same css rule on each page. Create a common styles file and import it wherever necessary

hiteshparashar4 commented on views/cart.hbs:
```diff
@@ -0,0 +1,49 @@
+            Promo code can be applied on payment page
+        </span>
+        <button>Proceed to Checkout </button>
+    </footer>
```
Handlebars allows you to create a common layout or a template file in which you can insert partial html like footer header etc. This way you don't have to add footer header on every page

hiteshparashar4 commented on static/styles/products.css:
```diff
@@ -0,0 +1,142 @@
+        width:30%;
+        display: flex;
+        flex-direction: column;
+        background-color: #dcdcdcde;
```
use scss variables

VaishaliRamesh-ng committed:
Add files via upload

VaishaliRamesh-ng committed:
Add files via upload

Commit before demo, with functionality for Home, products, login register and carts page.

VaishaliRamesh-ng committed:
Add files via upload
